chain/network/src/peer/codec.rs

```diff
@@ -79,6 +95,7 @@ impl Decoder for Codec {
     type Error = Error;
 
     fn decode(&mut self, buf: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
+        self.buf_size_metric.set(buf.len() as i64);
```

Comment:
Ok, I got lost a bit. You are reporting the "number of bytes to read" rather than buffer capacity. Is that intentional?

Reply:
So, broadly, what I want is to know the current `.len()` of buffers at any given point in time (for tracking memory usage, `.capacity()` would be more correct, but I think "how fast our peers drain the data" is more useful stat). 

The *right* way to do that would be to report the `.len` of the buffer every time it changes, but that's tricky, as the code is spread out across two crates. But luckely the buffer is passed into the Codec every time it changes, so measuring the size of the buffer every time it passes through the codec seems close enough (and that's what we already effectively are doing through the EncoderCallback infra anyway). 

The high order bit here is that we use `.set` API, rather than `.inc/.dec`.